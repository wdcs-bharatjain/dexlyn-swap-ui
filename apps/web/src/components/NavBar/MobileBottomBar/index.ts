export { MobileBottomBar, MobileBottomBarLegacy } from './MobileBottomBar'
export { TDPActionTabs } from './TDPActionTabs'
