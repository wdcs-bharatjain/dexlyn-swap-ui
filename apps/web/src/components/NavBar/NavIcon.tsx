import { t } from 'i18n'
import { ReactNode } from 'react'
import styled from 'styled-components'

const Container = styled.div<{ $size: number; $isActive: boolean }>`
  width: ${({ $size }) => $size}px;
  height: ${({ $size }) => $size}px;
  color: ${({ $isActive, theme }) => ($isActive ? theme.neutral1 : theme.neutral2)};
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-align: center;
  background-color: transparent;
  border: none;
  cursor: pointer;
  border-radius: 16px;
  transition: 250;
  z-index: 1;
  transform: translateY(-1px) translateX(4px);
`

interface NavIconProps {
  children: ReactNode
  size?: number
  isActive?: boolean
  label?: string
  onClick: () => void
}

export const NavIcon = ({
  children,
  isActive = false,
  size = 40,
  label = t('common.navigationButton'),
  onClick,
}: NavIconProps) => {
  return (
    <Container $size={size} $isActive={isActive} onClick={onClick} aria-label={label}>
      {children}
    </Container>
  )
}
