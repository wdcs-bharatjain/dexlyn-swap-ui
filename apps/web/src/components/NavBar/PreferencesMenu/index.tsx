import { NavDropdown, NavDropdownDefaultWrapper } from 'components/NavBar/NavDropdown/index'
import { CurrencySettings } from 'components/NavBar/PreferencesMenu/Currency'
import { LanguageSettings } from 'components/NavBar/PreferencesMenu/Language'
import { PreferenceSettings } from 'components/NavBar/PreferencesMenu/Preferences'
import { PreferencesView } from 'components/NavBar/PreferencesMenu/shared'
import { useCallback, useState } from 'react'
import { Popover } from 'ui/src'
import { Global } from 'ui/src/components/icons'

export function PreferenceMenu() {
  const [settingsView, setSettingsView] = useState<PreferencesView>(PreferencesView.SETTINGS)
  const onExitMenu = useCallback(() => setSettingsView(PreferencesView.SETTINGS), [setSettingsView])

  return (
    <Popover placement="bottom" stayInFrame allowFlip onOpenChange={onExitMenu}>
      <Popover.Trigger padding={8}>
        <Global size={20} color="$neutral2" cursor="pointer" />
      </Popover.Trigger>
      <NavDropdown width={310}>
        <NavDropdownDefaultWrapper>
          {settingsView === PreferencesView.SETTINGS && (
            <PreferenceSettings setSettingsView={(view: PreferencesView) => setSettingsView(view)} />
          )}
          {settingsView === PreferencesView.LANGUAGE && <LanguageSettings onExitMenu={onExitMenu} />}
          {settingsView === PreferencesView.CURRENCY && <CurrencySettings onExitMenu={onExitMenu} />}
        </NavDropdownDefaultWrapper>
      </NavDropdown>
    </Popover>
  )
}
