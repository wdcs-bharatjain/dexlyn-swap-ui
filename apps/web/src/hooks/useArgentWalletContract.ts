import { useAccount } from 'hooks/useAccount'
import { useContract } from 'hooks/useContract'
import useIsArgentWallet from 'hooks/useIsArgentWallet'
import ArgentWalletContractABI from 'uniswap/src/abis/argent-wallet-contract.json'
import { ArgentWalletContract } from 'uniswap/src/abis/types'

export function useArgentWalletContract(): ArgentWalletContract | null {
  const account = useAccount()
  const isArgentWallet = useIsArgentWallet()
  return useContract(
    isArgentWallet ? account.address : undefined,
    ArgentWalletContractABI,
    true
  ) as ArgentWalletContract
}
