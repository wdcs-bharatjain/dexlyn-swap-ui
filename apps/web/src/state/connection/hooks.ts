export { default as useCurrencyBalance, useCurrencyBalances, useTokenBalance } from 'lib/hooks/useCurrencyBalance'
