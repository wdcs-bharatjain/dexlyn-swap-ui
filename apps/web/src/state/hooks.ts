import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux'
import store from 'state/index'

export const useAppDispatch = () => useDispatch<typeof store.dispatch>()
export const useAppSelector: TypedUseSelectorHook<ReturnType<typeof store.getState>> = useSelector
