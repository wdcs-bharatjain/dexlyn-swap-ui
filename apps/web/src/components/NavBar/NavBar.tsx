import { Bag } from 'components/NavBar/Bag'
import { ChainSelector } from 'components/NavBar/ChainSelector'
import { CompanyMenu } from 'components/NavBar/CompanyMenu'
import { GetTheAppButton } from 'components/NavBar/DownloadApp/GetTheAppButton'
import { PreferenceMenu } from 'components/NavBar/PreferencesMenu'
import { useTabsVisible } from 'components/NavBar/ScreenSizes'
import { SearchBar } from 'components/NavBar/SearchBar'
import { Tabs } from 'components/NavBar/Tabs/Tabs'
import Row from 'components/Row'
import Web3Status from 'components/Web3Status'
import { useScreenSize } from 'hooks/screenSize'
import { useAccount } from 'hooks/useAccount'
import { useIsLandingPage } from 'hooks/useIsLandingPage'
import { useIsNftPage } from 'hooks/useIsNftPage'
import { useProfilePageState } from 'nft/hooks'
import { ProfilePageStateType } from 'nft/types'
import styled from 'styled-components'
import { BREAKPOINTS } from 'theme'
import { Z_INDEX } from 'theme/zIndex'

const Nav = styled.nav`
  padding: 0px 12px;
  width: 100%;
  height: ${({ theme }) => theme.navHeight}px;
  z-index: ${Z_INDEX.sticky};
  display: flex;
  align-items: center;
  justify-content: center;
`
const NavContents = styled.div`
  max-width: ${({ theme }) => `${theme.breakpoint.xxxl}px`};
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: center;
  flex: 1 auto 1;
`
const Left = styled(Row)`
  display: flex;
  align-items: center;
  gap: 12px;
  @media screen and (max-width: ${BREAKPOINTS.sm}px) {
    gap: 8px;
  }
`
const SearchContainer = styled.div`
  display: flex;
  flex: 1;
  flex-shrink: 1;
  justify-content: center;
  align-self: center;
  align-items: flex-start;
  height: 42px;
`

export const RefreshedNavbar = () => {
  const isNftPage = useIsNftPage()
  const isLandingPage = useIsLandingPage()
  const sellPageState = useProfilePageState((state) => state.state)
  const isSmallScreen = !useScreenSize()['sm']
  const areTabsVisible = useTabsVisible()
  const collapseSearchBar = !useScreenSize()['lg']
  const account = useAccount()
  const NAV_SEARCH_MAX_HEIGHT = 'calc(100vh - 30px)'

  return (
    <Nav>
      <NavContents>
        <Left gap="12px" wrap="nowrap">
          <CompanyMenu />
          {areTabsVisible && <Tabs />}
        </Left>

        <SearchContainer data-cy="center-search-container">
          {!collapseSearchBar && <SearchBar maxHeight={NAV_SEARCH_MAX_HEIGHT} fullScreen={isSmallScreen} />}
        </SearchContainer>

        <Row gap="12px" justify="flex-end" alignSelf="flex-end">
          {collapseSearchBar && <SearchBar maxHeight={NAV_SEARCH_MAX_HEIGHT} fullScreen={isSmallScreen} />}
          {isNftPage && sellPageState !== ProfilePageStateType.LISTING && <Bag />}
          {isLandingPage && !isSmallScreen && <GetTheAppButton showIcons={false} />}
          {!account.isConnected && <PreferenceMenu />}
          {!isNftPage && <ChainSelector />}
          <Web3Status />
        </Row>
      </NavContents>
    </Nav>
  )
}
