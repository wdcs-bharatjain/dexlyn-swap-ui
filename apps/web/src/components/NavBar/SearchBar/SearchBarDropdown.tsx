import { InterfaceSectionName, NavBarSearchTypes } from '@uniswap/analytics-events'
import { ChainId } from '@uniswap/sdk-core'
import Badge from 'components/Badge'
import Column from 'components/Column'
import { ChainLogo } from 'components/Logo/ChainLogo'
import { useRecentlySearchedAssets } from 'components/NavBar/SearchBar/RecentlySearchedAssets'
import { SkeletonRow, SuggestionRow } from 'components/NavBar/SearchBar/SuggestionRow'
import Row from 'components/Row'
import { SuspendConditionally } from 'components/Suspense/SuspendConditionally'
import { SuspenseWithPreviousRenderAsFallback } from 'components/Suspense/SuspenseWithPreviousRenderAsFallback'
import { BACKEND_NOT_YET_SUPPORTED_CHAIN_IDS } from 'constants/chains'
import { SearchToken } from 'graphql/data/SearchTokens'
import useTrendingTokens from 'graphql/data/TrendingTokens'
import { useTrendingCollections } from 'graphql/data/nft/TrendingCollections'
import { useAccount } from 'hooks/useAccount'
import { useDisableNFTRoutes } from 'hooks/useDisableNFTRoutes'
import { useIsNftPage } from 'hooks/useIsNftPage'
import { Trans } from 'i18n'
import { GenieCollection } from 'nft/types'
import { useEffect, useMemo, useState } from 'react'
import { Clock, TrendingUp } from 'react-feather'
import { useLocation } from 'react-router-dom'
import styled from 'styled-components'
import { ThemedText } from 'theme/components'
import { HistoryDuration, SafetyLevel } from 'uniswap/src/data/graphql/uniswap-data-api/__generated__/types-and-hooks'
import { InterfaceSearchResultSelectionProperties } from 'uniswap/src/features/telemetry/types'
import { useTrace } from 'utilities/src/telemetry/trace/TraceContext'

const SearchBarDropdownContainer = styled(Column)<{ $loading: boolean }>`
  width: 100%;
  backdrop-filter: blur(60px);
  overflow-y: scroll;
  transition: 125;
  opacity: ${({ $loading }) => ($loading ? '0.3' : '1')};
`
const DropdownHeader = styled(Row)`
  color: ${({ theme }) => theme.neutral2};
  line-height: 20px;
  padding: 4px 16px;
  font-weight: 500;
  font-size: 14px;
`
const NotFoundContainer = styled.div`
  font-size: 14px;
  line-height: 20px;
  font-weight: 500;
  color: ${({ theme }) => theme.neutral2};
  padding: 4px 16px;
`

interface SearchBarDropdownSectionProps {
  toggleOpen: () => void
  suggestions: (GenieCollection | SearchToken | undefined)[]
  header: JSX.Element
  headerIcon?: JSX.Element
  hoveredIndex?: number
  startingIndex: number
  setHoveredIndex: (index: number | undefined) => void
  isLoading?: boolean
  eventProperties: InterfaceSearchResultSelectionProperties
}

function SearchBarDropdownSection({
  toggleOpen,
  suggestions,
  header,
  headerIcon = undefined,
  hoveredIndex,
  startingIndex,
  setHoveredIndex,
  isLoading,
  eventProperties,
}: SearchBarDropdownSectionProps) {
  return (
    <Column gap="4px" data-testid="searchbar-dropdown">
      <DropdownHeader gap="8px">
        {headerIcon ? headerIcon : null}
        {header}
      </DropdownHeader>
      <Column gap="4">
        {suggestions.map((suggestion, index) =>
          isLoading || !suggestion ? (
            <SkeletonRow key={index} />
          ) : (
            <SuggestionRow
              key={suggestion.address}
              suggestion={suggestion}
              isHovered={hoveredIndex === index + startingIndex}
              setHoveredIndex={setHoveredIndex}
              toggleOpen={toggleOpen}
              index={index + startingIndex}
              eventProperties={{
                ...eventProperties,
                position: index + startingIndex,
                selected_search_result_name: suggestion.name,
                selected_search_result_address: suggestion.address,
              }}
            />
          )
        )}
      </Column>
    </Column>
  )
}

function isKnownToken(token: SearchToken) {
  return token.project?.safetyLevel == SafetyLevel.Verified || token.project?.safetyLevel == SafetyLevel.MediumWarning
}

const ChainComingSoonBadge = styled(Badge)`
  align-items: center;
  background-color: ${({ theme }) => theme.surface2};
  color: ${({ theme }) => theme.neutral2};
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  opacity: 1;
  padding: 8px;
  margin: 16px 16px 4px;
  width: calc(100% - 32px);
  gap: 8px;
`

interface SearchBarDropdownProps {
  toggleOpen: () => void
  tokens: SearchToken[]
  collections: GenieCollection[]
  queryText: string
  hasInput: boolean
  isLoading: boolean
}

export function SearchBarDropdown(props: SearchBarDropdownProps) {
  const { isLoading } = props
  const account = useAccount()
  const showChainComingSoonBadge =
    account.chainId && BACKEND_NOT_YET_SUPPORTED_CHAIN_IDS.includes(account.chainId) && !isLoading

  return (
    <SearchBarDropdownContainer $loading={isLoading}>
      <SuspenseWithPreviousRenderAsFallback>
        <SuspendConditionally if={isLoading}>
          <SearchBarDropdownContents {...props} />
        </SuspendConditionally>
      </SuspenseWithPreviousRenderAsFallback>
      {showChainComingSoonBadge && account.chainId && (
        <ChainComingSoonBadge>
          <ChainLogo chainId={account.chainId} size={20} />
          <ThemedText.BodySmall color="neutral2" fontSize="14px" fontWeight="400" lineHeight="20px">
            <ComingSoonText chainId={account.chainId} />
          </ThemedText.BodySmall>
        </ChainComingSoonBadge>
      )}
    </SearchBarDropdownContainer>
  )
}

function SearchBarDropdownContents({
  toggleOpen,
  tokens,
  collections,
  queryText,
  hasInput,
}: SearchBarDropdownProps): JSX.Element {
  const [hoveredIndex, setHoveredIndex] = useState<number | undefined>(0)
  const { data: searchHistory } = useRecentlySearchedAssets()
  const shortenedHistory = useMemo(() => searchHistory?.slice(0, 2) ?? [...Array<SearchToken>(2)], [searchHistory])
  const { pathname } = useLocation()
  const isNFTPage = useIsNftPage()
  const isTokenPage = pathname.includes('/explore')
  const shouldDisableNFTRoutes = useDisableNFTRoutes()
  const account = useAccount()

  const { data: trendingCollections, loading: trendingCollectionsAreLoading } = useTrendingCollections(
    3,
    HistoryDuration.Day
  )

  const formattedTrendingCollections = useMemo(() => {
    return !trendingCollectionsAreLoading
      ? trendingCollections
          ?.map((collection) => ({
            ...collection,
            collectionAddress: collection.address,
            floorPrice: collection.floor,
            stats: {
              total_supply: collection.totalSupply,
              one_day_change: collection.floorChange,
              floor_price: collection.floor,
            },
          }))
          .slice(0, isNFTPage ? 3 : 2) ?? []
      : [...Array<GenieCollection>(isNFTPage ? 3 : 2)]
  }, [trendingCollections, isNFTPage, trendingCollectionsAreLoading])

  const { data: trendingTokenData } = useTrendingTokens(account.chainId)

  const trendingTokensLength = !isNFTPage ? 3 : 2
  const trendingTokens = useMemo(
    () => trendingTokenData?.slice(0, trendingTokensLength) ?? [...Array<SearchToken>(trendingTokensLength)],
    [trendingTokenData, trendingTokensLength]
  )

  const totalSuggestions = hasInput
    ? tokens.length + collections.length
    : Math.min(shortenedHistory.length, 2) +
      (isNFTPage ? formattedTrendingCollections?.length ?? 0 : 0) +
      (!isNFTPage ? trendingTokens?.length ?? 0 : 0)

  // Navigate search results via arrow keys
  useEffect(() => {
    const keyDownHandler = (event: KeyboardEvent) => {
      if (event.key === 'ArrowUp') {
        event.preventDefault()
        if (!hoveredIndex) {
          setHoveredIndex(totalSuggestions - 1)
        } else {
          setHoveredIndex(hoveredIndex - 1)
        }
      } else if (event.key === 'ArrowDown') {
        event.preventDefault()
        if (hoveredIndex && hoveredIndex === totalSuggestions - 1) {
          setHoveredIndex(0)
        } else {
          setHoveredIndex((hoveredIndex ?? -1) + 1)
        }
      }
    }

    document.addEventListener('keydown', keyDownHandler)

    return () => {
      document.removeEventListener('keydown', keyDownHandler)
    }
  }, [toggleOpen, hoveredIndex, totalSuggestions])

  const hasVerifiedCollection = collections.some((collection) => collection.isVerified)
  const hasKnownToken = tokens.some(isKnownToken)
  const showCollectionsFirst = isNFTPage && (hasVerifiedCollection || !hasKnownToken)

  const trace = useTrace({ section: InterfaceSectionName.NAVBAR_SEARCH })

  const eventProperties = {
    total_suggestions: totalSuggestions,
    query_text: queryText,
    ...trace,
  }

  const tokenSearchResults =
    tokens.length > 0 ? (
      <SearchBarDropdownSection
        hoveredIndex={hoveredIndex}
        startingIndex={showCollectionsFirst ? collections.length : 0}
        setHoveredIndex={setHoveredIndex}
        toggleOpen={toggleOpen}
        suggestions={tokens}
        eventProperties={{
          suggestion_type: NavBarSearchTypes.TOKEN_SUGGESTION,
          ...eventProperties,
        }}
        header={<Trans i18nKey="common.tokens" />}
      />
    ) : (
      <NotFoundContainer>
        <Trans i18nKey="tokens.noneFound" />
      </NotFoundContainer>
    )

  const collectionSearchResults =
    collections.length > 0 ? (
      <SearchBarDropdownSection
        hoveredIndex={hoveredIndex}
        startingIndex={showCollectionsFirst ? 0 : tokens.length}
        setHoveredIndex={setHoveredIndex}
        toggleOpen={toggleOpen}
        suggestions={collections}
        eventProperties={{
          suggestion_type: NavBarSearchTypes.COLLECTION_SUGGESTION,
          ...eventProperties,
        }}
        header={<Trans i18nKey="nft.collections" />}
      />
    ) : (
      <NotFoundContainer>
        <Trans i18nKey="nft.noneFound" />
      </NotFoundContainer>
    )

  return hasInput ? (
    // Empty or Up to 8 combined tokens and nfts
    <Column gap="20">
      {showCollectionsFirst ? (
        <>
          {collectionSearchResults}
          {tokenSearchResults}
        </>
      ) : (
        <>
          {tokenSearchResults}
          {collectionSearchResults}
        </>
      )}
    </Column>
  ) : (
    // Recent Searches, Trending Tokens, Trending Collections
    <Column gap="20">
      {shortenedHistory.length > 0 && (
        <SearchBarDropdownSection
          hoveredIndex={hoveredIndex}
          startingIndex={0}
          setHoveredIndex={setHoveredIndex}
          toggleOpen={toggleOpen}
          suggestions={shortenedHistory}
          eventProperties={{
            suggestion_type: NavBarSearchTypes.RECENT_SEARCH,
            ...eventProperties,
          }}
          header={<Trans i18nKey="search.recent" />}
          headerIcon={<Clock width={20} height={20} />}
          isLoading={!searchHistory}
        />
      )}
      {!isNFTPage && (
        <SearchBarDropdownSection
          hoveredIndex={hoveredIndex}
          startingIndex={shortenedHistory.length}
          setHoveredIndex={setHoveredIndex}
          toggleOpen={toggleOpen}
          suggestions={trendingTokens}
          eventProperties={{
            suggestion_type: NavBarSearchTypes.TOKEN_TRENDING,
            ...eventProperties,
          }}
          header={<Trans i18nKey="common.popularTokens" />}
          headerIcon={<TrendingUp width={20} height={20} />}
          isLoading={!trendingTokenData}
        />
      )}
      {Boolean(!isTokenPage && !shouldDisableNFTRoutes) && (
        <SearchBarDropdownSection
          hoveredIndex={hoveredIndex}
          startingIndex={shortenedHistory.length + (isNFTPage ? 0 : trendingTokens?.length ?? 0)}
          setHoveredIndex={setHoveredIndex}
          toggleOpen={toggleOpen}
          suggestions={formattedTrendingCollections as unknown as GenieCollection[]}
          eventProperties={{
            suggestion_type: NavBarSearchTypes.COLLECTION_TRENDING,
            ...eventProperties,
          }}
          header={<Trans i18nKey="nft.popularCollections" />}
          headerIcon={<TrendingUp width={20} height={20} />}
          isLoading={trendingCollectionsAreLoading}
        />
      )}
    </Column>
  )
}

function ComingSoonText({ chainId }: { chainId: ChainId }) {
  switch (chainId) {
    case ChainId.AVALANCHE:
      return <Trans i18nKey="search.avalancheComing" />
    default:
      return null
  }
}
