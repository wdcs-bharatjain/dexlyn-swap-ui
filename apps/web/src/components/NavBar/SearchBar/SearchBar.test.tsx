import { SearchBar } from 'components/NavBar/SearchBar'
import { useScreenSize } from 'hooks/screenSize/useScreenSize'
import { useDisableNFTRoutes } from 'hooks/useDisableNFTRoutes'
import { useTranslation } from 'i18n/useTranslation'
import { useIsNavSearchInputVisible } from 'nft/hooks/useIsNavSearchInputVisible'
import { mocked } from 'test-utils/mocked'
import { render, screen } from 'test-utils/render'

jest.mock('hooks/useDisableNFTRoutes')
jest.mock('hooks/screenSize/useScreenSize')
jest.mock('nft/hooks/useIsNavSearchInputVisible')

describe('disable nft on searchbar', () => {
  beforeEach(() => {
    mocked(useScreenSize).mockReturnValue({
      xs: true,
      sm: true,
      md: true,
      lg: true,
      xl: false,
      xxl: false,
      xxxl: false,
      navSearchInputVisible: false,
      navDropdownMobileDrawer: true,
    })
    mocked(useIsNavSearchInputVisible).mockReturnValue(true)
  })

  it('should render text with nfts', () => {
    mocked(useDisableNFTRoutes).mockReturnValue(false)
    const { container } = render(<SearchBar />)
    expect(container).toMatchSnapshot()
    const { t } = useTranslation()
    expect(screen.queryByPlaceholderText(t('common.searchTokensNFT'))).toBeVisible()
  })
  it('should render text without nfts', () => {
    mocked(useDisableNFTRoutes).mockReturnValue(true)
    const { container } = render(<SearchBar />)
    expect(container).toMatchSnapshot()
    const { t } = useTranslation()
    expect(screen.queryByPlaceholderText(t('common.searchTokens'))).toBeVisible()
  })
})
