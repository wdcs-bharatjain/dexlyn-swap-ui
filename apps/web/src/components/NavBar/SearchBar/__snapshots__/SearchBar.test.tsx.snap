// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`disable nft on searchbar should render text with nfts 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.c5 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c0 {
  position: relative;
}

.c1 {
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: auto 1fr;
  grid-template-areas: 'input' 'dropdown';
  max-height: 100vh;
  z-index: 1000;
}

.c7 {
  background-color: #22222212;
  color: #7D7D7D;
  padding: 0px 8px;
  width: 20px;
  height: 20px;
  border-radius: 4px;
  font-size: 12px;
  font-weight: 535;
  line-height: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  opacity: 0.6;
  -webkit-backdrop-filter: blur(60px);
  backdrop-filter: blur(60px);
}

.c3 {
  grid-area: input;
  background-color: #F9F9F9;
  border: 1px solid #22222212;
  min-width: 280px;
  max-width: 400px;
  width: 100vw;
  padding: 8px 16px;
  border-radius: 20px;
  height: 40px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3:hover {
  background-color: #F5F5F5;
}

.c6 {
  width: 20px;
  height: 20px;
}

@media screen and (max-width:396px) {
  .c3 {
    border: none;
  }
}

<div>
  <span
    class="t_light is_inversed _dsp_contents"
  >
    <span
      class=" t_light _dsp_contents is_Theme"
    >
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <div
            class="c2 c3"
          >
            <div
              class="c4 c5"
            >
              <div
                class="c6"
              >
                <svg
                  fill="none"
                  height="20px"
                  stroke="#7D7D7D"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="20px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="11"
                    cy="11"
                    r="8"
                  />
                  <line
                    x1="21"
                    x2="16.65"
                    y1="21"
                    y2="16.65"
                  />
                </svg>
              </div>
              <span
                class="_dsp_contents"
              >
                <input
                  autocapitalize="sentences"
                  autocomplete="on"
                  autocorrect="on"
                  class="css-textinput-11aywtz r-placeholderTextColor-6taxm2 is_Input font_body _borderTopColor-0hover-1394778429 _borderRightColor-0hover-1394778429 _borderBottomColor-0hover-1394778429 _borderLeftColor-0hover-1394778429 _borderTopColor-0focus-1452587353 _borderRightColor-0focus-1452587353 _borderBottomColor-0focus-1452587353 _borderLeftColor-0focus-1452587353 _outlineColor-0focus-visible-834034279 _outlineWidth-0focus-visible-2px _outlineStyle-0focus-visible-solid _fontFamily-299667014 _fontWeight-233016078 _fontSize-229441158 _lineHeight-222976511 _pr-1481558307 _pl-1481558307 _height-10037 _borderTopLeftRadius-1881200688 _borderTopRightRadius-1881200688 _borderBottomRightRadius-1881200688 _borderBottomLeftRadius-1881200688 _borderTopWidth-1px _borderRightWidth-1px _borderBottomWidth-1px _borderLeftWidth-1px _outlineWidth-0px _color-675002279 _borderTopColor-2122800589 _borderRightColor-2122800589 _borderBottomColor-2122800589 _borderLeftColor-2122800589 _backgroundColor-733175080 _minWidth-0px _borderBottomStyle-solid _borderTopStyle-solid _borderLeftStyle-solid _borderRightStyle-solid _width-10037"
                  dir="auto"
                  placeholder="Search tokens and NFT collections"
                  rows="1"
                  spellcheck="true"
                  style="--placeholderTextColor: #7D7D7D;"
                  tabindex="0"
                  value=""
                />
              </span>
              <div
                class="c7"
              >
                /
              </div>
            </div>
          </div>
        </div>
      </div>
    </span>
  </span>
</div>
`;

exports[`disable nft on searchbar should render text without nfts 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.c5 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c0 {
  position: relative;
}

.c1 {
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: auto 1fr;
  grid-template-areas: 'input' 'dropdown';
  max-height: 100vh;
  z-index: 1000;
}

.c7 {
  background-color: #22222212;
  color: #7D7D7D;
  padding: 0px 8px;
  width: 20px;
  height: 20px;
  border-radius: 4px;
  font-size: 12px;
  font-weight: 535;
  line-height: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  opacity: 0.6;
  -webkit-backdrop-filter: blur(60px);
  backdrop-filter: blur(60px);
}

.c3 {
  grid-area: input;
  background-color: #F9F9F9;
  border: 1px solid #22222212;
  min-width: 280px;
  max-width: 400px;
  width: 100vw;
  padding: 8px 16px;
  border-radius: 20px;
  height: 40px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3:hover {
  background-color: #F5F5F5;
}

.c6 {
  width: 20px;
  height: 20px;
}

@media screen and (max-width:396px) {
  .c3 {
    border: none;
  }
}

<div>
  <span
    class=" _dsp_contents"
  >
    <span
      class=" t_light _dsp_contents is_Theme"
    >
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <div
            class="c2 c3"
          >
            <div
              class="c4 c5"
            >
              <div
                class="c6"
              >
                <svg
                  fill="none"
                  height="20px"
                  stroke="#7D7D7D"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="20px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="11"
                    cy="11"
                    r="8"
                  />
                  <line
                    x1="21"
                    x2="16.65"
                    y1="21"
                    y2="16.65"
                  />
                </svg>
              </div>
              <span
                class="_dsp_contents"
              >
                <input
                  autocapitalize="sentences"
                  autocomplete="on"
                  autocorrect="on"
                  class="css-textinput-11aywtz r-placeholderTextColor-6taxm2 is_Input font_body _borderTopColor-0hover-1394778429 _borderRightColor-0hover-1394778429 _borderBottomColor-0hover-1394778429 _borderLeftColor-0hover-1394778429 _borderTopColor-0focus-1452587353 _borderRightColor-0focus-1452587353 _borderBottomColor-0focus-1452587353 _borderLeftColor-0focus-1452587353 _outlineColor-0focus-visible-834034279 _outlineWidth-0focus-visible-2px _outlineStyle-0focus-visible-solid _fontFamily-299667014 _fontWeight-233016078 _fontSize-229441158 _lineHeight-222976511 _pr-1481558307 _pl-1481558307 _height-10037 _borderTopLeftRadius-1881200688 _borderTopRightRadius-1881200688 _borderBottomRightRadius-1881200688 _borderBottomLeftRadius-1881200688 _borderTopWidth-1px _borderRightWidth-1px _borderBottomWidth-1px _borderLeftWidth-1px _outlineWidth-0px _color-675002279 _borderTopColor-2122800589 _borderRightColor-2122800589 _borderBottomColor-2122800589 _borderLeftColor-2122800589 _backgroundColor-733175080 _minWidth-0px _borderBottomStyle-solid _borderTopStyle-solid _borderLeftStyle-solid _borderRightStyle-solid _width-10037"
                  dir="auto"
                  placeholder="Search tokens"
                  rows="1"
                  spellcheck="true"
                  style="--placeholderTextColor: #7D7D7D;"
                  tabindex="0"
                  value=""
                />
              </span>
              <div
                class="c7"
              >
                /
              </div>
            </div>
          </div>
        </div>
      </div>
    </span>
  </span>
</div>
`;
